apps/yapms/src/routes/app/[country]/[map]/[year]/initialize/ApplyPanZoom.ts

```diff
@@ -3,13 +3,36 @@ import panzoom, { type PanZoom } from 'panzoom';
 function applyPanZoom(mapBind: HTMLDivElement): PanZoom | undefined {
 	const svg = mapBind.querySelector<SVGElement>('svg'); //We can do this since the only child of node is the map svg itself.
 	if (svg) {
-		return panzoom(svg, {
+		const panzoomInstance = panzoom(svg, {
 			minZoom: 0.5,
 			maxZoom: 100,
 			smoothScroll: false,
 			autocenter: true,
 			zoomDoubleClickSpeed: 1
 		});
+		if (svg.hasAttribute('auto-border-stroke-width')) {
+			const initStroke = Number(svg.getAttribute('auto-border-stroke-width'));
+			const strokeUpper = svg.hasAttribute('auto-border-stroke-width-limit')
+				? Number(svg.getAttribute('auto-border-stroke-width-limit'))
+				: NaN;
+			adjustStroke(svg, initStroke, strokeUpper, panzoomInstance.getTransform().scale);
```

Comment:
This looks good, if you could just change the NaN to a null, and check for null instead I think that will be good. Generally NaN is reserved for errors in math computations, and not for control logic.


---

Nit: If you wanna handle all possible errors from user inputs, I think zod would be good here.

Users can pass in gibberish to generate NaN or "Infinity" to generate infinity.

By using something like z.number().finite().positive() you can verify that the user enters correct numbers.

![image](https://github.com/yapms/yapms/assets/13600696/83573628-d124-4521-8075-f6d2b51042f9)

Reply:
Oh I like this, will implement in a way that if initstroke is invalid, will return early.